Makefile

```diff
@@ -26,9 +25,16 @@ DOWNSTREAM_IMAGE_VERSION := "v3\.2\.0\.20"
 
 upstream:
 
-	curl -sL -O ${SOURCE}
+	sed -i ${UPSTREAM_NAME}.spec \
+	    -e "s|^Source0:\(\s\+\)\(.*\)$ |Source0:\1\%{name}-\%{version}\.tar\.gz|" \
+	    -e "s|^Version:\(\s\+\)\(.*\)$ |Version:\1${VERSION}|"
+	mkdir -p ${UPSTREAM_NAME}-${VERSION}
+	tar cf - --exclude=${UPSTREAM_NAME}-${VERSION} --exclude=.git . | (cd ${UPSTREAM_NAME}-${VERSION} && tar xvf - )
+	tar -cvf ${UPSTREAM_NAME}-${VERSION}.tar.gz ${UPSTREAM_NAME}-${VERSION} && \
+	    rm -fr ${UPSTREAM_NAME}-${VERSION}
```

Comment:
@praveenkumar I am having hard time understanding the sed commands and their purpose. Can these be simplified or can we add some comments which will help understand them?